### PR TITLE
Fix panic if any osinfo calls fail

### PR DESCRIPTION
**Pull request by adjackura**
Add more error details to WIndows inventory calls
Record osinfo details wven if some functions return errors.
Reorder windows osinfo functions in order of most likely to not return error.

### CODE REVIEW

File: osinfo/osinfo_windows.go [hunk 1]
```diff
@@ -127,45 +127,40 @@ func getKernelInfo() (string, string, error) {
...
-	hn, err := os.Hostname()
-	if err != nil {
-		return oi, fmt.Errorf("os.Hostname() error: %v", err)
+	var ops []win32OperatingSystem
+	query := "SELECT Caption, Version FROM Win32_OperatingSystem"
+	if err := wmi.Query(query, &ops); err != nil {
+		return oi, fmt.Errorf("wmi.Query(%q) error: %v", query, err)
 	}
-	oi.Hostname = hn
+	oi.LongName = ops[0].Caption
```
Comment (ryanwe) [inline osinfo/osinfo_windows.go:162]: Would it be safer to include bounds checking here?

Comment (adjackura) [reply osinfo/osinfo_windows.go:162]: Probably a good idea of the calling function changes.

**Pull request merged**